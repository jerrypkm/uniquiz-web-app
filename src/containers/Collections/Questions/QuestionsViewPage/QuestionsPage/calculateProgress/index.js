const calculateProgress = (itemsCount, points) => {
  return itemsCount && points
    ? `${Math.round((points / itemsCount) * 100)}% Complete`
    : "0% Complete";
};

export default calculateProgress;
