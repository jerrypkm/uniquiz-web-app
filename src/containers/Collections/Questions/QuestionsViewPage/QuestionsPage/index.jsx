/**
 * Course page contains layout styling for the page, progress status, and a list of questions
 * @param {Array} questions - List of questions passed to this component from the db
 * @param {Object} courseDetails - Contains details about the course we are looking at, this is queried from the db
 * @param {Number} points - Number of points the user has for this course
 */

import React, { lazy } from "react";

import calculateProgress from "./calculateProgress";
import Footer from "components/shared/Footer";
import Grid from "@material-ui/core/Grid";
import MoPage from "components/library/MoPage";
import MoPointsGroup from "components/library/MoPointsGroup";
import MoButtonIcon from "components/library/MoButtonIcon";
import NewItemCard from "components/shared/CardList/NewItemCard";

const CardList = lazy(() => import("components/shared/CardList"));

const QuestionsPage = ({
  questions,
  courseDetails,
  handleOnClick,
  isLoading,
  itemOptions,
  isAdmin,
  points,
}) => {
  if (isLoading) {
    return null;
  }
  const IconComponent = () => (
    <MoButtonIcon editIcon={true} handleIconClick={(e) => handleOnClick(e)} />
  );
  return (
    <MoPage
      title={courseDetails?.data?.title}
      IconComponent={isAdmin && IconComponent}
    >
      <Grid container spacing={4}>
        <Grid item xs={12} sm={12} md={12} lg={12}>
          {
            <MoPointsGroup
              authUser={itemOptions?.authUser}
              points={points}
              progress={calculateProgress(questions?.length, points)}
            />
          }
        </Grid>
        <CardList
          isAdmin={isAdmin}
          items={questions}
          itemOptions={itemOptions}
        />
        {isAdmin && (
          <Grid item xs={12} sm={12} md={4} lg={4}>
            <NewItemCard
              title={itemOptions?.newItem?.title}
              url={itemOptions?.newItem?.url}
            />
          </Grid>
        )}
      </Grid>
      <Footer />
    </MoPage>
  );
};

export default QuestionsPage;
