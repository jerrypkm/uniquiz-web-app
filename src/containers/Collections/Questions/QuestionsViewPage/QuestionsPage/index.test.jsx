import React from "react";

import ShallowRenderer from "react-test-renderer/shallow";
import QuestionsPage from ".";

const renderer = new ShallowRenderer();

describe("QuestionsPage Component", () => {
  test("should match the snapshot", () => {
    renderer.render(<QuestionsPage />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
