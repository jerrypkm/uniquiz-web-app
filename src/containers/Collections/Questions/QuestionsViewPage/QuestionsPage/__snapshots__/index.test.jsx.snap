// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionsPage Component should match the snapshot 1`] = `
<ForwardRef(WithStyles)>
  <ForwardRef(WithStyles)
    container={true}
    spacing={4}
  >
    <ForwardRef(WithStyles)
      item={true}
      lg={12}
      md={12}
      sm={12}
      xs={12}
    >
      <MoPointsGroup
        progress="0% Complete"
      />
    </ForwardRef(WithStyles)>
    <UNDEFINED />
  </ForwardRef(WithStyles)>
  <ForwardRef(WithStyles) />
</ForwardRef(WithStyles)>
`;
